qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -352,6 +352,12 @@ def _assemble(self, circuits, parameter_binds=None, **run_options):
             circuits, optypes, run_options = self._assemble_noise_model(
                 circuits, optypes, **run_options)
 
+            # Remove metadata from circuits
+            metadata_list = [circuit.metadata for circuit in circuits]
+            for circuit in circuits:
+                if circuit.metadata:
+                    circuit.metadata.clear()
```

Comment:
The way this is written, `metadata_list` only contains references to the metadata dictionaries, so this line actually deletes the metadata. Instead of clearing the dictionaries, set them to `None`.